packages/graphql-codegen-typescript-operations/src/visitor.ts

```diff
@@ -138,17 +145,27 @@ export class TypeScriptDocumentsVisitor extends BaseDocumentsVisitor<
         this.config.immutableTypes,
         this.config.namespacedImportName,
         enumsNames,
-        this.config.enumPrefix,
+        false,
         this.config.enumValues,
         true,
         this.config.inlineCommonTypes,
+        this.usedEnums,
       ),
     );
     this._declarationBlockConfig = {
       ignoreExport: this.config.noExport,
     };
   }
 
+  private getExportNames(): string[] {
+    const exportNames: string[] = Array.from(this.usedTypes);
+
+    if (!this.config.inlineCommonTypes) {
+      exportNames.push(...Array.from(this.usedEnums));
```

Comment:
Nit: Array.from is redundant
```suggestion
      exportNames.push(...this.usedEnums);
```